src/main.rs

```diff
@@ -86,6 +88,24 @@ fn run() -> i32 {
         None => root,
     };
 
+    let output_ext = match matches.opt_str("f") {
+        Some(s) => match s.to_lowercase().as_ref() {
+            "png" => "png",
+            "pam" => "pam",
+            _ => {
+                eprintln!("Invalid image format specified, defaulting to png");
```

Comment:
I'd rather abort on invalid parameters.

Reply:
If we do this,
```rust
    match s.to_lowercase().as_ref() {
            "png" => "png",
            "pam" => "pam",
    }
```

will look silly. Instead we should just check that it's a known one (just making sure to point this out so we don't have that silly case).

Reply:
@Lonami We have to match _ because of [E0004](https://doc.rust-lang.org/error-index.html#E0004).
Aborting is good, though I don't like the `eprint` `usage` `return` pattern that's used elsewhere in the main function. Maybe usage should take an optional message to print?

Reply:
Yes, probably. My take:
```let (output_ext, output_format) = match matches.opt_str("f").map(String::to_lowercase) {
    Some("png") | None => ("png", image::ImageOutputFormat::PNG),
    Some("pam") => ("pam", image::ImageOutputFormat::PNM(PNMSubtype::ArbitraryMap)),
    _ => {
        eprintln!("Invalid image format specified");
        return 1;
    },
}
```
Combines both blocks into one, shorter and leaner.

Reply:
My take:
```rust
let output_ext = matches.opt_str("f").unwrap_or("png").to_lowercase();
let output_format = match output_ext {
    "png" => image::ImageOutputFormat::PNG,
    "pam" => image::ImageOutputFormat::PNM(PNMSubtype::ArbitraryMap,
    _ => {
        eprintln!("Invalid image format specified");
        return 1;
    }
}
```

Reply:
That's even better actually.

Reply:
I need to learn rust lol